src/frontend/js/pages/TeacherDashboardContractsLayout/hooks/useDownloadContractArchive/index.tsx

```diff
@@ -0,0 +1,123 @@
+import { useEffect, useState } from 'react';
+import useContractArchive from 'hooks/useContractArchive';
+import { useContracts } from 'hooks/useContracts';
+import { ContractState } from 'types/Joanie';
+import { PER_PAGE } from 'settings';
+import useCheckContractArchiveExists from '../useCheckContractArchiveExists';
+import {
+  storeContractArchiveId,
+  unstoreContractArchiveId,
+  getStoredContractArchiveId,
+  isStoredContractArchiveIdExpired,
+} from './contractArchiveLocalStorage';
+
+export enum ContractDownloadStatus {
+  IDDLE = 'iddle',
+  PENDING = 'pending',
+  READY = 'ready',
+}
+
+interface UseTeacherContractsBulkDownloadProps {
+  organizationId: string;
+}
+
+const useDownloadContractArchive = ({ organizationId }: UseTeacherContractsBulkDownloadProps) => {
```

Comment:
Overall comment: I think having the tests in a separate commit makes the review much more difficult as we cannot read the tests AND the source code at the same time, splited by well identified feature.

IMO a commit = a feature, including its tests, which ease the understanding by reading the tests themselves.